### DISCUSSION TITLE
how to use different .env files for different environments, such as development, test/staging or production

### DISCUSSION THREAD

**Discussion by dhavalveera** [vercel/next.js#38821] [Help]
Hello,

Nextjs identifies the `.env.development` for dev environment & `.env.production` for the production environment, but I've 1 question

we've 3 phased for the Web App
 - development
 - staging (for testing)
 - production

now I wanted to use `.env.development` for the dev environment only when it's in development & pushed to the server
`.env.test` ---- for staging environment only and
`.env.production` for -- production environment only

anyone can help me or guide me on how I can achieve this?

**Comment by tofuwu-dev** [comment]
Do you use vercel or others for hosting website? Or are you gonna use those env files on local??

**Reply by dhavalveera** [reply]
> Do you use vercel or others for hosting website? Or are you gonna use those env files on local??

for some website - `Vercel` and for others using `AWS EC2`

**Reply by tofuwu-dev** [reply]
if you use vercel, you should be able to set envs on vercel each other (for production, for development/staging)

**Reply by dhavalveera** [reply]
> if you use vercel, you should be able to set envs on vercel each other (for production, for development/staging)

how? but I'm using `.env` files for -- production, development & staging & connecting those `.env` files to `next.config.js`

**Reply by tofuwu-dev** [reply]
![image](https://user-images.githubusercontent.com/98631674/179972651-f775c5b6-99a6-4608-b644-0020f85d1b0e.png)
![image](https://user-images.githubusercontent.com/98631674/179972733-b2420e94-c6ac-4a20-b7f2-82f4ebde0301.png)

production is on master(main) branch and you can set branch for development(i.e. staging branch)
Vercel is also providing preview link

Could you plz quote your next config file, here?

**Reply by dhavalveera** [reply]
```js
next.config.js

module.export = {
   env: {....}
}
```


also, how to use different `.env` files on AWS EC2 - which is a Linux Server??

**Reply by tofuwu-dev** [reply]
> ```js
> next.config.js
> 
> module.export = {
>    env: {....}
> }
> ```
> 
> also, how to use different `.env` files on AWS EC2 - which is a Linux Server??

May I trouble you can quote different env files in next js file? , just want to know how to configure different env in next config for proudction/development...
And not sure AWS. sorry.. you can deploy your website using vercel and buy domain :), That is it, I know just this.

**Reply by dhavalveera** [reply]
I don't know how to configure different `.env` files such as

`.env.staging`
`.env.development`
`.env.production`

I don't know how to do it, and also I don't know how to inform Nextjs to use `.env.staging` for Test/Staging env

**Reply by tofuwu-dev** [reply]
these files will be using on local
i.e. yarn dev -> using .env.development, yarn start -> using .env.production
So just add env variables on vercel each other

**Reply by dhavalveera** [reply]
but we've 3 stages:

1. dev env
2. staging env
3. production env


and we host the website on AWS EC2 and based on the environment, I want to use the related `.env` file

can you help me with that? I don't need help regarding Vercel.
